### PR TITLE
#166828867 Fix errors on code

### PR DESCRIPTION
**Pull request by cvjude**
**What does this PR do?**
Fixes the bugs on the backend and on the Docs

**Description of Task to be completed?**
> Remove password input from /auth/users/{role}
> GET /auth/getusers change this description to "Get all users")
> /auth/verifyemail . add ID to query string
> /auth/profile remove id from route
> /author/favourtie/{id} Add route to Author section
> /books add tags field
> /books/request REMOVE categoryID
> /books/request CHANGE patronID in docs to patronId.
> change order of docs, /books/lend should come before /books/receive
> /books/request change statusCode for returning a book to 200 not 400.

**How should this be manually tested?**
Clone the repo and run npm install to download all packages, then use npm run test to test the endpoint, POSTMAN can also be used to test.

**Any background context you want to provide?**
None

**What are the relevant pivotal tracker stories?**<a href = "https://www.pivotaltracker.com/story/show/166828867"> #166828867</a>

**Screenshots (if appropriate)**
None

### CODE REVIEW

File: server/controllers/authorController.js [hunk 1]
```diff
@@ -61,6 +62,51 @@ class AuthorController {
     
     return util.successStatus(res, 200, 'Author deleted successfully');
   }
+
+  static async favouriteAnAuthor(req, res) {
```
Comment (emp-daisy) [inline server/controllers/authorController.js]: THought we already merged this?

File: server/controllers/userController.js [hunk 2]
```diff
@@ -334,51 +337,6 @@ class UserController {
       favoriteBooks: updatedUser.favoriteBooks
     })
   }
-
```
Comment (emp-daisy) [inline server/controllers/userController.js:337]: Oh I see. It was moved to another file

File: server/docs/hellobooks_api_doc.yaml [hunk 3]
```diff
@@ -552,21 +570,15 @@ paths:
                   type: number
                 damaged:
                   type: string
```
Comment (emp-daisy) [inline server/docs/hellobooks_api_doc.yaml:574]: Isn't this a boolean?

File: server/docs/hellobooks_api_doc.yaml [hunk 4]
```diff
@@ -945,12 +943,6 @@ paths:
         - bearerAuth: []
       summary: Get user profile
       description: User can get other user profiles including theirs
-      parameters:
-        - in: query
-          name: id
-          schema:
-            type: integer
-          required: true
```
Comment (emp-daisy) [inline server/docs/hellobooks_api_doc.yaml:953]: I'm guessing this param was to get other user's profile too.
Therefore this should be here but not required

Comment (emp-daisy) [reply server/docs/hellobooks_api_doc.yaml:953]: @encodedBicoding

Comment (encodedBicoding) [reply server/docs/hellobooks_api_doc.yaml:953]: That is true, it should be false. @cvjude give the required property value of false

**Review by Hadeneekeh — APPROVED** [review]

**Review by encodedBicoding — DISMISSED** [review]

File: server/routes/authorRoute.js [hunk 5]
```diff
@@ -10,5 +10,12 @@ authorRoute.get('/', Authenticate.isLoggedIn, AuthorController.listAuthor);
 authorRoute.get('/:id', AuthorController.getAuthor);
```
Comment (Hadeneekeh) [inline server/routes/authorRoute.js]: This route was supposed to be protected.

**Review by Hadeneekeh — CHANGES_REQUESTED** [review]

File: server/controllers/authorController.js (hunk 1 above)
Comment (cvjude) [reply server/controllers/authorController.js]: I dont know what happened there
I just rebased from develop

File: server/docs/hellobooks_api_doc.yaml (hunk 3 above)
Comment (cvjude) [reply server/docs/hellobooks_api_doc.yaml:574]: I implemented it as a string

File: server/docs/hellobooks_api_doc.yaml (hunk 4 above)
Comment (cvjude) [reply server/docs/hellobooks_api_doc.yaml:953]: The userId was implemented in the request query.

File: server/routes/authorRoute.js (hunk 5 above)
Comment (cvjude) [reply server/routes/authorRoute.js]: For the Admin alone?

File: server/docs/hellobooks_api_doc.yaml (hunk 4 above)
Comment (emp-daisy) [reply server/docs/hellobooks_api_doc.yaml:953]: What do you mean @cvjude

Comment (emp-daisy) [reply server/docs/hellobooks_api_doc.yaml:953]: So I am guessing the route serves two purposes, if the ID is provided, it gets the user with given ID profile else it gets the logged in user profile

File: server/docs/hellobooks_api_doc.yaml (hunk 3 above)
Comment (emp-daisy) [reply server/docs/hellobooks_api_doc.yaml:574]: I see. That wasn't the best option but could pass

File: server/routes/authorRoute.js (hunk 5 above)
Comment (emp-daisy) [reply server/routes/authorRoute.js]: For logged in users

File: server/controllers/userController.js [hunk 6]
```diff
@@ -289,37 +289,40 @@ class UserController {
...
-        favoriteQuote: user.favoriteQuote,
-        favoriteBooks: user.favoriteBooks
-      })
+
+    let user;
+    if(userId){
+      user = await returnUserObj(userId);
+    }
+    else user = await returnUserObj(loggedinUser.id);
+    return util.successStatus(res, 200, 'profile retrieved successfully', {
```
Comment (emp-daisy) [inline server/controllers/userController.js:309]: @encodedBicoding You didn't account for when the profile doesn't exist

File: server/docs/hellobooks_api_doc.yaml (hunk 4 above)
Comment (cvjude) [reply server/docs/hellobooks_api_doc.yaml:953]: Yes, that is what it does

File: server/controllers/userController.js (hunk 6 above)
Comment (cvjude) [reply server/controllers/userController.js:309]: I'll get on it now.

**Review by emp-daisy — APPROVED** [review]

**Pull request merged**